app/src/main/java/kiwi/root/an2linuxclient/activities/MainSettingsActivity.java

```diff
@@ -19,32 +18,49 @@
 import android.content.pm.PackageInfo;
 import android.content.pm.PackageManager;
 import android.graphics.BitmapFactory;
+import android.graphics.Color;
 import android.net.ConnectivityManager;
 import android.net.NetworkInfo;
+import android.os.Build;
+import android.os.Bundle;
 import android.preference.Preference;
 import android.preference.PreferenceFragment;
 import android.preference.PreferenceManager;
+import android.support.annotation.RequiresApi;
+import android.support.v4.app.ActivityCompat;
 import android.support.v4.app.NotificationCompat;
 import android.support.v4.app.NotificationManagerCompat;
 import android.support.v7.app.AlertDialog;
 import android.support.v7.app.AppCompatActivity;
-import android.os.Bundle;
 import android.text.Html;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.Toast;
-
-import java.util.List;
-
 import kiwi.root.an2linuxclient.R;
 import kiwi.root.an2linuxclient.crypto.KeyGeneratorService;
 import kiwi.root.an2linuxclient.data.MobileServer;
 import kiwi.root.an2linuxclient.data.ServerDatabaseHandler;
 import kiwi.root.an2linuxclient.data.WifiServer;
+import kiwi.root.an2linuxclient.utils.AN2LinuxService;
 import kiwi.root.an2linuxclient.utils.ConnectionHelper;
 
+import java.util.List;
+
 public class MainSettingsActivity extends AppCompatActivity {
+    private final static int HIDING_NOTIFICATION_ID = 2;
+    private final static String INFORMATION_CHANNEL_ID = "AN2LinuxInformationChannel";
+
+    private boolean useForegroundService = false;
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        if (useForegroundService){
+            startService(new Intent(this, AN2LinuxService.class));
+        }
+    }
 
```

Comment:
I think we could use the SharedPreferences that's automatically created here for if the user have enabled useForegroundService or not.

Like this instead:

```
boolean useForegroundService = PreferenceManager.getDefaultSharedPreferences(this)
        .getBoolean(getString(R.string.preference_enable_service), false);
if (useForegroundService){
    startService(new Intent(this, AN2LinuxService.class));
}
```

And also I think the code in `onStart`:
```
if (useForegroundService){
    startService(new Intent(this, AN2LinuxService.class));
}
```
is not needed, `onResume` is enough as it will be called every time after `onCreate` and `onStart` including the first time.